tests/ci/create-commit.sh

```diff
@@ -0,0 +1,378 @@
+IMAGE_KEY=${IMAGE_KEY}
+TEMPDIR=${TEMPDIR}
+EOF
+
+# Get the compose log.
+get_compose_log () {
+    COMPOSE_ID=$1
+    LOG_FILE=osbuild-${ID}-${VERSION_ID}-${COMPOSE_ID}.log
+
+    # Download the logs.
+    composer-cli compose log "$COMPOSE_ID" | tee "$LOG_FILE" > /dev/null
+}
+
+# Get the compose metadata.
+get_compose_metadata () {
+    COMPOSE_ID=$1
+    METADATA_FILE=osbuild-${ID}-${VERSION_ID}-${COMPOSE_ID}.json
+
+    # Download the metadata.
+    composer-cli compose metadata "$COMPOSE_ID" > /dev/null
+
+    # Find the tarball and extract it.
+    TARBALL=$(basename "$(find . -maxdepth 1 -type f -name "*-metadata.tar")")
+    tar -xf "$TARBALL" -C "${TEMPDIR}"
+    rm -f "$TARBALL"
+
+    # Move the JSON file into place.
+    cat "${TEMPDIR}"/"${COMPOSE_ID}".json | jq -M '.' | tee "$METADATA_FILE" > /dev/null
+}
+
+# Build ostree image.
+build_image() {
+    blueprint_file=$1
+    blueprint_name=$2
+
+    # Prepare the blueprint for the compose.
+    greenprint "📋 Preparing blueprint"
+    composer-cli blueprints push "$blueprint_file"
+    composer-cli blueprints depsolve "$blueprint_name"
+
+    # Start the compose.
+    greenprint "🚀 Starting compose"
+    composer-cli --json compose start "$blueprint_name" "$IMAGE_TYPE" | tee "$COMPOSE_START"
+    COMPOSE_ID=$(jq -r '.build_id' "$COMPOSE_START")
+
+    # Wait for the compose to finish.
+    greenprint "⏱ Waiting for compose to finish: ${COMPOSE_ID}"
+    while true; do
+        composer-cli --json compose info "${COMPOSE_ID}" | tee "$COMPOSE_INFO" > /dev/null
+        COMPOSE_STATUS=$(jq -r '.queue_status' "$COMPOSE_INFO")
+
+        # Is the compose finished?
+        if [[ $COMPOSE_STATUS != RUNNING ]] && [[ $COMPOSE_STATUS != WAITING ]]; then
+            echo ; greenprint "🚀 Finished compose"
+            break
+        fi
+        echo -n "."
+
+        # Wait 30 seconds and try again.
+        sleep 5
+    done
+
+    # Capture the compose logs from osbuild.
+    greenprint "💬 Getting compose log and metadata"
+    get_compose_log "$COMPOSE_ID"
+    get_compose_metadata "$COMPOSE_ID"
+
+    # Did the compose finish with success?
+    if [[ $COMPOSE_STATUS != FINISHED ]]; then
+        echo "Something went wrong with the compose. 😢"
+        exit 1
+    fi
+}
+
+##################################################
+##
+## ostree image/commit installation
+##
+##################################################
+
+# Write a source for the Neptune app.
+tee "$SOURCE_FILE" > /dev/null << EOF
+id = "copr_neptune"
+name = "copr_neptune"
+type = "yum-baseurl"
+url = "https://download.copr.fedorainfracloud.org/results/pingou/qtappmanager-fedora/epel-8-${ARCH}"
+check_gpg = false
+check_ssl = false
+system = false
+EOF
+
+# Add COPR Neptune source
+greenprint "📝 Add COPR Neptune source"
+sudo composer-cli sources add "$SOURCE_FILE"
+sudo composer-cli sources list
+sudo composer-cli sources info copr_neptune
+
+# Write a blueprint for ostree image.
+tee "$BLUEPRINT_FILE" > /dev/null << EOF
```

Comment:
would it make sense to file an issue to the base-enablement team, so they provide the blueprint for installation instead of us hardcoding it into test?

Reply:
Yes, it does. This is how this is been done at the `ostree.sh` file for the [upstream](https://github.com/osbuild/osbuild-composer/blob/main/test/cases/ostree.sh#L247) and [downstream](https://github.com/virt-s1/rhel-edge/blob/main/ostree.sh#L304) tests for the `osbuild-composer`, but I prefer the idea of having a file we can track.

Extracting the blueprint is one of the next changes I wanted to do, but it's good idea to have the file sync by the base-enablement team.

BTW, I think we could clean up a bit the blueprint dependencies. I use the blueprint for the aarch64 demo, which has fewer dependencies than the amd64 demo and I remove anther package that we don't really need for the Neptune app.
The sorter the list, the better. Fewer packages to track, smaller image and less time to build.

Reply:
Just a question - what is this BLUEPRINT used for?

Reply:
@guy9050 Sorry, I didn't see the notification.

`$BLUEPRINT` is a file defined [here](https://github.com/osbuild/automotive-ci/blob/main/tests/ci/create-commit.sh#L17) which is used to tell the `composer-cli` the packages and other configs that we need in our image.
Here is where this is used:
https://github.com/osbuild/automotive-ci/blob/main/tests/ci/create-commit.sh#L61

The file has the list of packages that we need to build our image, so it makes sense what @thrix was suggestion about to have it in a separated file and maintain by the ones that know what the image looks like.